nemo/collections/nlp/models/language_modeling/megatron_base_prompt_learning_model.py

```diff
@@ -211,68 +170,25 @@ def init_prompt_encoder(self):
         new_task = self.new_tasks[0]
         total_virtual_tokens = self.task_templates[new_task]["total_virtual_tokens"]
 
-        encoder_type = PromptEncoderType(self.cfg.p_tuning.get("encoder_type", "mlp").lower())
+        encoder_type = PromptEncoderType(self.cfg.p_tuning.get("encoder_type", "tpmlp").lower())
         self.prompt_encoder = PromptEncoder(
-            encoder_type=PromptEncoderType(encoder_type),
+            encoder_type=encoder_type,
             total_virtual_tokens=total_virtual_tokens,
             token_dim=self.hidden_size,
-            hidden_size=self.cfg.p_tuning.get("encoder_hidden", 2048),
+            hidden_size=self.cfg.p_tuning.get("encoder_hidden", self.hidden_size // 2),
```

Comment:
is `self.hidden_size // 2` a good number to use?

Reply:
It's a better default, for when the base model has different hidden sizes.